cache-config/testing/ort-tests/t3c-lockfile_test.go

```diff
@@ -0,0 +1,113 @@
+package orttest
+
+/*
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import (
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/apache/trafficcontrol/cache-config/testing/ort-tests/tcdata"
+)
+
+func TestLockfile(t *testing.T) {
+	testName := "TestLockfile"
+	t.Logf("------------- Starting " + testName + " ---------------")
+	tcd.WithObjs(t, []tcdata.TCObj{
+		tcdata.CDNs, tcdata.Types, tcdata.Tenants, tcdata.Parameters,
+		tcdata.Profiles, tcdata.ProfileParameters, tcdata.Statuses,
+		tcdata.Divisions, tcdata.Regions, tcdata.PhysLocations,
+		tcdata.CacheGroups, tcdata.Servers, tcdata.Topologies,
+		tcdata.DeliveryServices}, func() {
+
+		hostName := "atlanta-edge-03"
+		const fileName = `records.config`
+
+		firstOut := ""
+		firstCode := 0
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			t.Logf("TestLockFile first t3c starting %v", time.Now())
+			firstOut, firstCode = t3cUpdateReload(hostName, "badass")
+			t.Logf("TestLockFile first t3c finished %v", time.Now())
+		}()
+
+		time.Sleep(time.Millisecond * 100) // sleep long enough to ensure the concurrent t3c starts
+		t.Logf("TestLockFile second t3c starting %v", time.Now())
+		out, code := t3cUpdateReload(hostName, "badass")
+		t.Logf("TestLockFile second t3c finished %v", time.Now())
+		if code != 0 {
+			t.Fatalf("second t3c apply badass failed: output '''%v''' code %v\n", out, code)
```

Comment:
I have two nits here:
- you don't need the `\n`, the logging makes sure it ends with a newline character for you
- using `%v` instead of a more specific formatting parameter for primitive types/`fmt.Stringer`s makes it harder to catch obscure issues like printing a memory address instead of the value stored there, so I encourage people to be more specific when possible. Not that you're printing pointer values on this particular line, or anything. Just a good practice IMO.

Reply:
I removed the newline.

I don't agree about `%v`. In my professional opinion, "stronger types" in format strings make the code more brittle and likely to break without actually gaining anything. Unless the format string is used for machine-read data, which is rare, there's little value in forcing it to break when the type changes. For pointers in particular, it's sometimes valuable to print the address, so it's valuable to make it explicit to readers that the pointer or value is being printed, by passing the pointer or value explicitly. Calling `fmt.Sprintf("%d", pointer)` will simply print it as decimal instead of hex, it neither fails `go vet` nor fails to compile.

Reply:
+1 on the nitpick (more specific types used in format strings). IMO it's tremendously more likely for a developer to accidentally print a pointer address when they really meant to print the actual value, than it is that a type changed and needs a changed format string in various places. Using specific types forces the developer to actually think about what is getting printed and help nudge them into things like dereferencing the pointer where needed. But, like you said, for things like printing pointers to ints, you can't always rely on the compiler or `go vet`.

That said, it is definitely a nitpick, which by definition, I wouldn't necessarily hold up the PR for.

Reply:
Changed.